pkg/agent/cniserver/server.go

```diff
@@ -461,6 +468,15 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	klog.Infof("CmdAdd for container %v succeeded", cniConfig.ContainerId)
 	// mark success as true to avoid rollback
 	success = true
+
+	if s.secNetworkEnabled {
```

Comment:
took care as suggested by @jianjuns (do not use feature gate all over the code. used bool variable to decide whether we need to cache the cni info or not)